tools/docker/Dockerfile

```diff
@@ -0,0 +1,231 @@
+# syntax=docker/dockerfile:experimental
+# Copyright 2018-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+# 	http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+
+#########################################
+#
+# BUILD IMAGES
+#
+#########################################
+
+
+# Common tools needed for the build stages ahead. The final test images do not inherit directly from here, so this bloat
+# is dropped in those final end-use images.
+FROM debian:stretch as build-base
+ENV PATH /bin:/usr/bin:/usr/local/bin:/sbin:/usr/sbin:/usr/local/sbin:/usr/lib/go/bin
+RUN --mount=type=cache,id=aptcache,target=/var/cache/apt \
```

Comment:
This is to share the apt cache among containers?  Is there a lot of value in doing this given that we're unconditionally running `apt update` anyway?